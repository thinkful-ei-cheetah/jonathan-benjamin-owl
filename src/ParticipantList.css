.Participant-list {
    margin: 0;
}

.avatar {
    height: 25px;
    margin-right: 5px;
}

.inSession {
    background-color: green;
}

.status {
    text-align: center;
}

.outOfSession {
    background-color: gray;
}

.status-symbol {
    border: 1px solid black;
    height: 10px;
    width: 10px;
    border-radius: 50%;
    display: inline-block;
    margin-left: 10px;
    margin-right: 15px;
}