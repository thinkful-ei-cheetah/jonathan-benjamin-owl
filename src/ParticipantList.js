import React, { Component } from 'react';
import './ParticipantList.css';


export default class ParticipantList extends Component {

  sessionStatus(status) {
    return status ? "inSession" : "outOfSession";
  }

  status(sessionStatus, stageStatus) {
    return !sessionStatus ? "left session" : (stageStatus ? "on stage" : "in session")
  }

  render() {
    return ( 
      <div className="Participant-list">
        {this.props.participants.map((person) =>
            <div> 
              <div className="item-row">
                <img className="avatar" src={person.avatar} alt="{person.name} avatar"/>
                <span>{person.name}</span>
              </div>
              <div className="item-row">
                <div className={'status-symbol ' + this.sessionStatus(person.inSession)}></div>   
                <span className="status">{this.status(person.inSession, person.onStage)}</span>
              </div>
            </div>      
        )}
      </div>
    );}
}