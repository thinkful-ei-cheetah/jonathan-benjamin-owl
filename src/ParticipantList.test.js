import React from 'react';
import ReactDOM from 'react-dom';
import ParticipantList from './ParticipantList.js';
import renderer from 'react-test-renderer';

const participants = [
    {
        id: 1,
        name: 'Koren Templeton',
        avatar:
            'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: true
    }
]

//this is the test case
it('renders without crashing', () => {
    // first create a DOM element to render the component into
    const div = document.createElement('div');

    //render the component, this is the actual test, if something is wrong it will fail here
    ReactDOM.render(<ParticipantList participants={participants}/>, div);

    //clean up code
    ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
        .create(<ParticipantList participants={participants}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
