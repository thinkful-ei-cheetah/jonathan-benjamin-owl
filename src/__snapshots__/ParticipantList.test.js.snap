// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Participant-list"
>
  <div>
    <div
      className="item-row"
    >
      <img
        alt="{person.name} avatar"
        className="avatar"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
      <span>
        Koren Templeton
      </span>
    </div>
    <div
      className="item-row"
    >
      <div
        className="status-symbol inSession"
      />
      <span
        className="status"
      >
        on stage
      </span>
    </div>
  </div>
</div>
`;
